### DISCUSSION TITLE
Decimal and thousand separator not working as aspected

### DISCUSSION THREAD

**Discussion by gtoniolo** [handsontable/hyperformula#1452] [API Questions]
I'm trying to sum two big number. The example is quite simple.

https://stackblitz.com/edit/handsontable-hyperformula-demos-zefdpc?file=src%2FhyperformulaConfig.js

But I always have zero. What I'm doing wrong?

**Comment by sequba** [comment]
Hi @gtoniolo, thanks for reaching out ;)

First, you have a typo in the name of the parameter. It should be `thousandSeparator`.

Second, according to the [documentation](https://hyperformula.handsontable.com/api/interfaces/configparams.html#thousandseparator), `thousandSeparator` must be different from [functionArgSeparator](https://hyperformula.handsontable.com/api/interfaces/configparams.html#functionargseparator), which is set to `,` by default.

Updated demo: https://stackblitz.com/edit/handsontable-hyperformula-demos-hfdzxw?file=src%2FhyperformulaConfig.js

Does it fix your issue?

**Reply by gtoniolo** [reply]
Hi @sequba. Thank you for your answare.

 Actually no. If I only add functionArgSeparator: ';' I have error as result. 
https://stackblitz.com/edit/handsontable-hyperformula-demos-zefdpc?file=src%2FhyperformulaConfig.js
In your example I see number without separator
![image](https://github.com/user-attachments/assets/2dd580a8-0bc8-4daa-ba0d-3c9ceccd7b56)

In mine there are separator
![image](https://github.com/user-attachments/assets/a45ac07e-7dc2-4742-bbf0-43c863df303a)

**Comment by sequba** [comment]
> If I only add functionArgSeparator: ';' I have error as result.

Because you also need to replace `,` with `;` in your data (change `=SUM(A1,B1)` to `=SUM(A1;B1)`).

> In your example I see number without separator

Because now it's correctly recognized as a number and displayed as such (previously, it was displayed as a string). If you want to adjust the way the demo displays numbers (separators, decimal digits, etc.), you must edit the function `renderTable` accordingly. HyperFormula itself returns raw JS numbers. It is up to the UI app to display the number in the right format.

**Comment by gtoniolo** [comment]
Thank you so much. I was writing thousandsSeparator with two S....
